### PR TITLE
630 add testing steps

### PR DESCRIPTION
**Pull request by ngaruko**
#630

### CODE REVIEW

File: content/en/contribute/code/workflow.md [hunk 1]
```diff
@@ -69,6 +69,22 @@ All features and bug fixes must have at least one unit test. All features must h
...
 
+### Acceptance Testing
+
+  1 . Assign it to yourself.
+  2. Install the PR branch to test against
+  3. If the issue fails AT then notify the original developer and move the issue back to "In progress".
+  4. Once the issue passes AT notify the original developer and move the issue to "Ready to merge"
+  5. The developer will then "Squash and Merge" the PR, delete the PR branch and, if a backport is required cherry-pick the merged commit back to the release branches it's required in.
+  6. Check if there is an e2e test for the feature, if aplicable.
+  7. Close the issue, which will move it to `Done`
```
Comment (garethbowen) [inline content/en/contribute/code/workflow.md:80]: I think this duplicates a lot of the information in the "AT in progress" section below. Could you have a look at merging the two? I think it fits better down below but there may be additional useful information in this write up.

File: content/en/contribute/code/workflow.md [hunk 2]
```diff
@@ -69,6 +69,22 @@ All features and bug fixes must have at least one unit test. All features must h
...
+
+
+
+### Release Testing
+We have a release [testing repository](https://github.com/medic/cht-release-testing) with major workflows. Most issues will have a step-by-step description or a link to some details for execution. At release tesing time, create a [project board](https://github.com/medic/cht-release-testing#generating-a-project-board-for-testing-a-release) with all issues labelled 'release'.
+Start testing following instuctions on the ticket. If the test passes, move it to `pass` and if it fails,move it to `fail`, raise an issue and notify the team. A decision will be made whether to halt the release (blocker) or schedule the fix for the next release.
```
Comment (garethbowen) [inline content/en/contribute/code/workflow.md:86]: Typo and made the pass/fail steps more clear:

```suggestion
Start testing following instructions on the ticket. If the test passes move it to `pass`. If it fails move it to `fail`, raise an issue and notify the team. A decision will be made whether to halt the release (blocker) or schedule the fix for the next release.
```

File: content/en/contribute/code/workflow.md [hunk 3]
```diff
@@ -69,6 +69,22 @@ All features and bug fixes must have at least one unit test. All features must h
...
+
```
Comment (garethbowen) [inline content/en/contribute/code/workflow.md:87]: It'd be good to include a section summarising the sorts of things that are tested in the release testing issues, for example: "The release testing suite covers a range of tests to check for any regressions, performance tests, and scalability tests." And then maybe link to the results of each step, so people can find more info? We should probably do a better job of publishing our performance testing results but that may be out of scope of this issue.

**Review by garethbowen — CHANGES_REQUESTED** [review]
Awesome start, thanks!

I've put a few comments inline.

File: content/en/contribute/code/workflow.md (hunk 1 above)
Comment (ngaruko) [reply content/en/contribute/code/workflow.md:80]: Agree- complete duplicate. Didn't notice that section - will remove this.

File: content/en/contribute/code/workflow.md (hunk 3 above)
Comment (ngaruko) [reply content/en/contribute/code/workflow.md:77]: Added a few more details on RT and scalability

File: content/en/contribute/code/workflow.md [hunk 4]
```diff
@@ -69,6 +69,12 @@ All features and bug fixes must have at least one unit test. All features must h
 
 Those are minimums. Our ultimate goal is to have fully-automated release testing, allowing for fast, confident delivery of completed code. If your work would be included in a release/regression test, create an e2e test for it. QA engineers will use that as a template of sorts to create additional automated tests at their discretion.
 
+### Release Testing
+The release testing suite covers a range of tests to check for any regressions, performance tests, and scalability tests. We have a release [testing repository](https://github.com/medic/cht-release-testing) with major workflows. Most issues will have a step-by-step description or a link to some details for execution. At release tesing time, create a [project board](https://github.com/medic/cht-release-testing#generating-a-project-board-for-testing-a-release) with all issues labelled 'release'.
```
Comment (garethbowen) [inline content/en/contribute/code/workflow.md:73]:
```suggestion
The release testing suite covers a range of tests to check for any regressions, performance tests, and scalability tests. We have a [release testing repository](https://github.com/medic/cht-release-testing) with major workflows. Most issues have a step-by-step description or a link to some details for execution. At release testing time, create a [project board](https://github.com/medic/cht-release-testing#generating-a-project-board-for-testing-a-release) with all issues labelled 'release'.
```

File: content/en/contribute/code/workflow.md [hunk 5]
```diff
@@ -69,6 +69,12 @@ All features and bug fixes must have at least one unit test. All features must h
...
+Start testing following instructions on the ticket. If the test passes move it to `pass`. If it fails move it to `fail`, raise an issue and notify the team. A decision will be made whether to halt the release (blocker) or schedule the fix for the next release. [Here](https://github.com/medic/cht-release-testing/projects/101) is a typical release test project with results in the [comment section](https://github.com/medic/cht-release-testing/issues/190#issuecomment-1032646259) on individual tickets.
+As part of release testing, there is a [scalablity test suite ](https://github.com/medic/cht-core/tree/master/tests/scalability) to test the scalability of CouchDB so we can check what our limits are and ensure we don't regress on performance. Initially it only tests replication but should be extended to test other APIs. The JMeter tests are run on an [AWS](https://github.com/medic/cht-core/tree/master/tests/scalability#running-tests-on-aws) install and a report is saved in an S3 bucket. We then compare results from previous runs and save a JTL file containing the results of the test run in `cht-core/tests/scalability/previous_results`.
```
Comment (garethbowen) [inline content/en/contribute/code/workflow.md:75]: The scalability test hits API so it tests basically everything to do with replication from API on down to the DB.

```suggestion
As part of release testing, there is a [scalability test suite](https://github.com/medic/cht-core/tree/master/tests/scalability) to test the scalability of the CHT server so we can check what our limits are and ensure we don't regress on performance. Initially it only tests replication but should be extended to test other APIs. The JMeter tests are run on an [AWS](https://github.com/medic/cht-core/tree/master/tests/scalability#running-tests-on-aws) install and a report is saved in an S3 bucket. We then compare results from previous runs and save a JTL file containing the results of the test run in the [cht-core repository](https://github.com/medic/cht-core/tree/master/tests/scalability/previous_results).
```

**Review by garethbowen — CHANGES_REQUESTED** [review]
Nice one! Some small recommendations inline. I think we should also mention the performance smoke testing, even though it's not very reliable, it's important that people know that we do monitor client side performance!

**Review by garethbowen — APPROVED** [review]
Awesome, thanks!

This PR is fine as is, but something to think about when you're back is whether we make our performance testing and results more public like we do with the scalability suite.

**Pull request merged**